src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1303,14 +1279,31 @@ private bool InternalSetTaskParameter(TaskPropertyInfo parameter, IList paramete
 
             if (LogTaskInputs && !_taskLoggingContext.LoggingService.OnlyLogCriticalEvents)
             {
-                // If the type is a list, we already logged the parameters
-                if (!(parameterValue is IList))
+                IList parameterValueAsList = parameterValue as IList;
+                bool legacyBehavior = !ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_12);
```

Comment:
This is just paranoia, right? The behavior of anything that's not `TaskParameterEvent` enlightened should be identical to what it was before?

(To be clear I endorse and support paranoia!)

Reply:
> The behavior of anything that's not TaskParameterEvent enlightened should be identical to what it was before?

The PR is switching task parameter logging from a mix of text + structured to all structured. Technically it is a breaking change for anything that depends on the text logging, so the answer to your question is no. The potential break seems to be unlikely but not impossible.

An alternative approach would be to double-log for now, and remove the text logging some time later but it a) increases binlog size and b) makes the viewer show duplicated parameters so we would need to patch the viewer first, and wait some time before enabling double-logging.

Reply:
> Technically it is a breaking change for anything that depends on the text logging

this is what I don't understand. Does the message text of the new structured message not exactly match the prior textual message?

Reply:
Oh, sorry, my statement is confusing. The new structured message converts to an identical string. So users won't see any difference in logs. The breaking potential applies only to loggers that depend on the exact type of the event. That's what you meant by _"anything that's not TaskParameterEvent enlightened"_, I get it now!

Reply:
I have merged the PR to unblock @f-alizada.